java/tests/integrationtests/com/google/idea/blaze/java/sync/projectstructure/JdksTest.java

```diff
@@ -222,6 +222,42 @@ public void testChooseDifferentJdkIfNotProvidedByProvider() {
     assertThat(chosenSdk).isNotEqualTo(currentJdk7);
   }
 
+  @Test
+  public void testChooseSameVersionAsPreviewLanguageLevel() {
+    Sdk jdk11 = getUniqueMockJdk(LanguageLevel.JDK_11);
+    Sdk jdk17 = getUniqueMockJdk(LanguageLevel.JDK_17);
+
+    registerJdkProvider(
+        ImmutableMap.of(
+            LanguageLevel.JDK_11, jdk11,
+            LanguageLevel.JDK_17, jdk17));
+
+    setJdkTable(jdk11, jdk17);
+
+    Sdk chosenSdk = Jdks.chooseOrCreateJavaSdk(jdk11, LanguageLevel.JDK_17_PREVIEW);
+    assertThat(chosenSdk).isNotEqualTo(jdk11);
+    assertThat(chosenSdk).isEqualTo(jdk17);
+  }
+
+  @Test
+  public void testChooseNoJdkIfNoMatchingPreviewVersionInAvailable() {
+    Sdk jdk8 = getUniqueMockJdk(LanguageLevel.JDK_1_8);
+    Sdk jdk11 = getUniqueMockJdk(LanguageLevel.JDK_11);
+
+    registerJdkProvider(
+        ImmutableMap.of(
+            LanguageLevel.JDK_1_8, jdk8,
+            LanguageLevel.JDK_11, jdk11
+        ));
+
+    setJdkTable(jdk8, jdk11);
+
+    Sdk chosenSdk = Jdks.chooseOrCreateJavaSdk(jdk11, LanguageLevel.JDK_17_PREVIEW);
+    // chosenSdk may be non-null when finding a local, system-installed sdk that provides the preview version
```

Comment:
I tried this locally with jdk17 installed and [`createJdk`](https://github.com/bazelbuild/intellij/blob/d8d6ddcb34fb122a8f00c45c0927d023fe300def/java/src/com/google/idea/blaze/java/sync/projectstructure/Jdks.java#L103) failed with `Unable to create JDK from path '{PATH_TO_MY_JDK17}'`, maybe this is not allowed during tests. I think it will be better to not depend on the environment on which the test is run.

Reply:
This test was trying to test the case when no jdk could be found but it looks like which can trigger lookup of a jdk on the local machine which is bad like you noted. I replaced this test with another test that choses an sdk based when multiple language levels are provided.